File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario3/id_20T2000Test.py
# id_20_Test.java
from scenario3.id_20 import find_closest_elements

import pytest

class id_20_Test:
    """
    Test class of id_20.py.
    It contains ten unit test cases for the :func:`scenario3.id_20.find_closest_elements`.
    """


    def test_find_closest_elements_1(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.2]) == (2.0, 2.2)

    def test_find_closest_elements_2(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.0]) == (2.0, 2.0)

    def test_find_closest_elements_3(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.1]) == (2.0, 2.1)

    def test_find_closest_elements_4(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.3]) == (2.0, 2.3)

    def test_find_closest_elements_5(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.4]) == (2.0, 2.4)

    def test_find_closest_elements_6(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.5]) == (2.0, 2.5)

    def test_find_closest_elements_7(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.6]) == (2.0, 2.6)

    def test_find_closest_elements_8(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.7]) == (2.0, 2.7)

    def test_find_closest_elements_9(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.8]) == (2.0, 2.8)

    def test_find_closest_elements_10(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.9]) == (2.0, 2.9)

# id_20_Test.py
from scenario3.id_20 import find_closest_elements

import pytest

class id_20_Test:
    """
    Test class of id_20.py.
    It contains ten unit test cases for the :func:`scenario3.id_20.find_closest_elements`.
    """
		
    def test_find_closest_elements_1(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.2]) == (2.0, 2.2)

    def test_find_closest_elements_2(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.0]) == (2.0, 2.0)

    def test_find_closest_elements_3(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.1]) == (2.0, 2.1)

    def test_find_closest_elements_4(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.3]) == (2.0, 2.3)

    def test_find_closest_elements_5(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.4]) == (2.0, 2.4)

    def test_find_closest_elements_6(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.5]) == (2.0, 2.5)

    def test_find_closest_elements_7(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.6]) == (2.0, 2.6)

    def test_find_closest_elements_8(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.7]) == (2.0, 2.7)

    def test_find_closest_elements_9(self):
        """
        Test case for the :func:`scenario3.id_20.find_closest_elements` function.
        """
        assert find_closest_elements([1.0, 2.0, 3.0, 4.0, 5.0, 2.8]) == (2.0, 2.8)

    def test_find_closest_elements_10(self):
        """
        Test case for the

File: RQ2_Prompt_Elements/StarCoder_Data/HumanEvalJava_output/scenario3/id_20T2000Test.java
// FindClosestElementsTest.java
package scenario3;

import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link FindClosestElements}.
* It contains ten unit test cases for the {@link FindClosestElements#findClosestElements(List)} method.
*/
class FindClosestElementsTest {

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of numbers of length at least two.
     */
    @Test
    void testFindClosestElements() {
        List<Double> numbers = List.of(1.0, 2.0, 3.9, 4.0, 5.0, 2.2);
        List<Double> expected = List.of(3.9, 4.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of numbers of length at least two.
     */
    @Test
    void testFindClosestElements2() {
        List<Double> numbers = List.of(1.0, 2.0, 5.9, 4.0, 5.0);
        List<Double> expected = List.of(5.0, 5.9);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of numbers of length at least two.
     */
    @Test
    void testFindClosestElements3() {
        List<Double> numbers = List.of(1.0, 2.0, 3.0, 4.0, 5.0, 2.2);
        List<Double> expected = List.of(2.0, 2.2);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of numbers of length at least two.
     */
    @Test
    void testFindClosestElements4() {
        List<Double> numbers = List.of(1.0, 2.0, 3.0, 4.0, 5.0, 2.0);
        List<Double> expected = List.of(2.0, 2.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of numbers of length at least two.
     */
    @Test
    void testFindClosestElements5() {
        List<Double> numbers = List.of(1.1, 2.2, 3.1, 4.1, 5.1);
        List<Double> expected = List.of(2.2, 3.1);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with an empty list.
     */
    @Test
    void testFindClosestElements6() {
        List<Double> numbers = List.of();
        List<Double> expected = List.of();
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of length one.
     */
    @Test
    void testFindClosestElements7() {
        List<Double> numbers = List.of(1.0);
        List<Double> expected = List.of(1.0, 1.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of length two.
     */
    @Test
    void testFindClosestElements8() {
        List<Double> numbers = List.of(1.0, 2.0);
        List<Double> expected = List.of(1.0, 2.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of length three.
     */
    @Test
    void testFindClosestElements9() {
        List<Double> numbers = List.of(1.0, 2.0, 3.0);
        List<Double> expected = List.of(1.0, 3.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }

    /**
     * Test case for the {@link FindClosestElements#findClosestElements(List)} method.
     * It tests the method with a list of length four.
     */
    @Test
    void testFindClosestElements10() {
        List<Double> numbers = List.of(1.0, 2.0, 3.0, 4.0);
        List<Double> expected = List.of(1.0, 4.0);
        assertEquals(expected, FindClosestElements.findClosestElements(numbers));
    }
}
